examples/tv_minimization_reconstruction_pdhg.py

```diff
@@ -0,0 +1,184 @@
+# data.
+#
+# Let $y$ denote the k-space data of the image $x_{\mathrm{true}}$ sampled with an acquisition model $A$
+# (Fourier transform, coil sensitivity maps, ...), i.e the forward problem is given as
+#
+# $ y = Ax_{\mathrm{true}} + n, $
+#
+# where $n$ describes complex Gaussian noise. When using TV-minimization as regularization method, an approximation of
+# $x_{\mathrm{true}}$ is obtained by minimizing the following functional $\mathcal{F}$:
+#
+# $ \mathcal{F}(x) = \frac{1}{2}||Ax - y||_2^2 + \lambda \| \nabla x \|_1, \quad \quad \quad (1)$
+#
+# where $\nabla$ is the discretized gradient operator.
+#
+# The minimization of the functional $\mathcal{F}$ is a non-trivial task due to the presence of the operator
+# $\nabla$ in the non-differentiable $\ell_1$-norm. A suitable algorithm to solve the problem is the
+# PDHG-algorithm [Chambolle \& Pock, JMIV 2011].
+#
+# PDHG is a method for solving problems of the form
+#
+# $ \min_x f(K(x)) + g(x)  \quad \quad \quad (2)$
+#
+# where $f$ and $g$ denote proper, convex, lower-semicontinous functionals and $K$ denotes a linear operator.
+#
+# PDHG essentially consists of three steps, which read as
+#
+# $z_{k+1} = \mathrm{prox}_{\sigma f^{\ast}}(z_k + \sigma K \bar{x}_k)$
+#
+# $x_{k+1} = \mathrm{prox}_{\tau g}(x_k - \tau K^H z_{k+1})$
+#
+# $\bar{x}_{k+1} = x_{k+1} + \theta(x_{k+1} - x_k)$,
+#
+# where $\mathrm{prox}$ denotes the proximal operator and $f^{\ast}$ denotes the convex conjugate of the
+# functional $f$, $\theta\in [0,1]$ and step sizes $\sigma, \tau$ such that $\sigma \tau < 1/L^2$, where
+# $L=\|K\|_2$ is the operator norm of the operator $K$.
+#
+# The first step is to recast problem (1) into the general form of (2) and then to apply the steps above
+# in an iterative fashion.
+#
+# A possible and intuitive (but unfortunately not efficient) identification is the following
+#
+# $f(x) = \lambda \| x\|_1,$
+#
+# $g(x) = \frac{1}{2}\|Ax  - y\|_2^2,$
+#
+# $K(x) = \nabla x.$
+#
+# However, although $\mathrm{prox}_{\sigma f^\ast}$ has a simple form, calculations show that
+# to be able to compute $\mathrm{prox}_{\tau g}$, one would need to solve a linear system at each
+# iteration. We will therefore use a way more efficient identification.
+#
+# In the following, we load some 2D radial MR data and PDHG to reconstruct it.
+
+# %%
+import mrpro
+
+# %% [markdown]
+# ##### Read-in the raw data
+
+# %%
+# Load in the Data from the ISMRMRD file
+kdata = mrpro.data.KData.from_file(data_file.name, mrpro.data.traj_calculators.KTrajectoryIsmrmrd())
+kdata.header.recon_matrix.x = 256
+kdata.header.recon_matrix.y = 256
+
+# %% [markdown]
+# ### Set up the operator $A$
+# Estimate coil sensitivity maps and density compensation function. Also run a direct (adjoint)
+# reconstruction and iterative SENSE as methods of comparison.
+
+# %%
+# Set up direct reconstruction class to estimate coil sensitivity maps, density compensation etc
```

Comment:
We don't need the direct reconstruction to get the dcf and csm but I guess it makes sense to reuse it for later to save time. Maybe reformulate to
```suggestion
# Set up direct reconstruction class. The estimate coil sensitivity maps and density compensation values can be reused later to save time.
```